llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -8619,6 +8619,11 @@ SDValue SITargetLowering::lowerWorkitemID(SelectionDAG &DAG, SDValue Op,
   if (MaxID == 0)
     return DAG.getConstant(0, SL, MVT::i32);
 
+  // It's undefined behavior if a function marked with the amdgpu-no-*
+  // attributes uses the corresponding intrinsic.
+  if (!Arg)
+    return DAG.getUNDEF(EVT::getIntegerVT(*DAG.getContext(), 32));
```

Comment:
EVT::getIntegerVT(*DAG.getContext(), 32) is an overly complicate way of doing just MVT::i32 as Is done in the 0 case. Plus this should really be using the original output type